### PR TITLE
Media Modal: <DetailFields /> component janitorial

### PR DESCRIPTION
**Pull request by retrofox**
### Janitorial PR

This PR is part of the #11093 issue.

---------

This PR implements janitorial process in the `<DetailField />` component.

Primary improvements:

- re-write component using ES class
- use localize() helper
- avoid binding function into a property
- rename methods trying to be more verbose
- several ESLint warnings, etc

### Testing

Open an item detail dialog.
Test to edit one of item fields: `Caption`, `Alt text` and `Description`
- You could take a look at the network tab paying attention to the requests triggered. It shouldn't trigger a request if the value doesn't change for instance
- Resize the viewport and scroll trying to hide an input and type there. It should be focused.
- Edit close and open the dialog once the changes are done.

Everything should be ok.

### CODE REVIEW

File: client/post-editor/media-modal/detail/detail-fields.jsx [hunk 1]
```diff
@@ -1,145 +1,151 @@
...
-	},
+	}
 
-	persistChange: function() {
+	persistChange() {
 		if ( ! this.props.site || ! this.state.modifiedItem ) {
 			return;
 		}
 
-		debug( 'Update media to %o', this.state.modifiedItem );
```
Comment (retrofox) [inline client/post-editor/media-modal/detail/detail-fields.jsx]: There is only one line bug in this component. It doesn't add too much information regarding the debugging process, especially keeping in mind that we can use the react tab.
Finally, I think that little by little, we are not using debug in our base code. Keep in mind this code is quite old according to the calypso history. Especially, this line is from the beginning.

But ... I could restore it if we consider that make sense keep it there. Not a big deal at all.

File: client/post-editor/media-modal/detail/detail-fields.jsx [hunk 2]
```diff
@@ -1,145 +1,151 @@
...
+import TrackInputChanges from 'components/track-input-changes';
+import EditorMediaModalFieldset from '../fieldset';
+
+class EditorMediaModalDetailFields extends Component {
+	static propTypes = {
+		site: PropTypes.object,
+		item: PropTypes.object
+	};
+
+	state = {};
```
Comment (retrofox) [inline client/post-editor/media-modal/detail/detail-fields.jsx]: I agree that the component could be written avoiding initiate the state with an empty object. But in this case, we should do a slightly deeper refactoring.

We should change this line, for instance.
```es6
if ( this.state.modifiedItem && attribute in this.state.modifiedItem ) {
  // ...
```

Thinking well it's doable in this PR, though. It's up to us.

Comment (gwwar) [inline client/post-editor/media-modal/detail/detail-fields.jsx]: Ah ok, up to you then on keeping it or updating usage. We can guard with lodash `get( state, [ 'modifiedItem', attribute ] )`

File: client/post-editor/media-modal/detail/detail-fields.jsx (hunk 1 above)
Comment (gwwar) [inline client/post-editor/media-modal/detail/detail-fields.jsx]: Feel free to remove then

**Comment by retrofox** [comment]
Changes from the last update/rebase:

* remove setting initial state 
* declare debounce function in the function constructor
* use lodash `get()`

@gwwar @iamtakashi it's ready for a new round. :-D

**Comment by iamtakashi** [comment]
I've tested this with Chrome, Mac. It seems clear out whatever I typed when I go to next field.

untitled

**Comment by gwwar** [comment]
I verified that flows in the media modal still work 👍  Let's run with this, but have the next PR fix the issue in the media section. Updating the title does update the item, but it doesn't look like it in the media details view.

title

**Pull request merged**